src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionMicroserviceVerticle.java

```diff
@@ -965,40 +975,137 @@ private void downloadFileAnnotation(RoutingContext event) {
-                        String filePath = fileInfo.getString("originalFilePath");
-                        //If the path is a directory, send error response
-                        File file = new File(filePath);
-                        if (!file.exists()) {
-                            response.setStatusCode(404);
-                            response.end("File for given Annotation does not exist");
-                            return;
-                        }
-                        if (file.isDirectory()) {
-                            response.setStatusCode(501);
-                            response.end("File Annotation of Unsupported File Type");
-                            return;
-                        }
-                        else {
-                            response.headers().set("Content-Type", "application/octet-stream");
-                            response.headers().set("Content-Disposition",
-                                    "attachment; filename=\"" + fileName + "\"");
-                            response.sendFile(filePath);
+            ImageRegionVerticle.GET_FILE_ANNOTATION_METADATA_EVENT,
+            Json.encode(annotationCtx), new Handler<AsyncResult<Message<JsonObject>>>() {
+                @Override
+                public void handle(AsyncResult<Message<JsonObject>> result) {
+                    if (result.failed()) {
+                        log.error(result.cause().getMessage());
+                        response.setStatusCode(404);
+                        response.end("Could not get annotation "
+                                    + request.getParam("annotationId"));
+                        return;
+                    }
+                    JsonObject fileInfo = result.result().body();
+                    String fileName = fileInfo.getString("originalFileName");
+                    String filePath = fileInfo.getString("originalFilePath");
+                    //If the path is a directory, send error response
+                    File file = new File(filePath);
+                    if (!file.exists()) {
+                        response.setStatusCode(404);
+                        response.end("File for given Annotation does not exist");
+                        return;
+                    }
+                    if (file.isDirectory()) {
+                        response.setStatusCode(501);
+                        response.end("File Annotation of Unsupported File Type");
+                        return;
+                    }
+                    else {
+                        response.headers().set("Content-Type", "application/octet-stream");
+                        response.headers().set("Content-Disposition",
+                                "attachment; filename=\"" + fileName + "\"");
+                        if (request.headers().contains("Range")) {
+                            String range = request.getHeader("Range");
```

Comment:
If a range is requested, the response code should be `206` for Partial Content, and the `content-range` response header should also be added. (Perhaps `sendfile` does this automatically?)

Also, open-ended byte ranges like `1024-` should be supported.